src/Features/Core/Portable/Snippets/SnippetProviders/AbstractConsoleSnippetProvider.cs

```diff
@@ -108,7 +98,31 @@ private async Task<TextChange> GenerateSnippetTextChangeAsync(Document document,
             return root.ReplaceNode(snippetExpressionNode, reformatSnippetNode);
         }
 
-        private static SyntaxNode? GetConsoleExpressionStatement(ISyntaxFactsService syntaxFacts, SyntaxNode root, int position)
+        protected override ImmutableArray<SnippetPlaceholder> GetPlaceHolderLocationsList(SyntaxNode node, ISyntaxFacts syntaxFacts, CancellationToken cancellationToken)
+        {
+            return ImmutableArray<SnippetPlaceholder>.Empty;
+        }
+
+        private static SyntaxToken? GetOpenParenToken(SyntaxNode node, ISyntaxFacts syntaxFacts)
+        {
+            var invocationExpression = node.DescendantNodes().Where(syntaxFacts.IsInvocationExpression).FirstOrDefault();
+            if (invocationExpression is null)
+            {
+                return null;
+            }
+
+            var argumentListNode = syntaxFacts.GetArgumentListOfInvocationExpression(invocationExpression);
+            if (argumentListNode is null)
+            {
+                return null;
+            }
+
+            syntaxFacts.GetPartsOfArgumentList(argumentListNode, out var openParenToken, out _, out _);
+
+            return openParenToken;
```

Comment:
i don't get why this method is nullable just to have the caller throw if it gets null.